src/contracts/core/DelegationManager.sol

```diff
@@ -480,6 +494,23 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
         return approverDigestHash;
     }
 
+    /** 
+     * @notice Returns 'true' if the staker can undelegate or  if the staker is already undelegated, and 'false' otherwise
+     * @dev A staker can only undelegate if they have no "active" shares in EigenLayer and are not themselves an operator
+     */
+    function stakerCanUndelegate(address staker) public view returns (bool) {
+        return (!stakerHasActiveShares(staker) && !isOperator(staker));
+
+    }
+
+    /**
+     * @notice Returns 'true' if `staker` has "active" shares in EigenLayer (i.e. the staker has shares which are currently in the StrategyManager
+     * or in the EigenPodManager + not in "undelegation limbo"), and returns 'false' otherwise.
+     */
+    function stakerHasActiveShares(address staker) public view returns (bool) {
+        return ((strategyManager.stakerStrategyListLength(staker) != 0) || !eigenPodManager.podOwnerHasNoDelegatedShares(staker));
```

Comment:
Double negation here:
```
!eigenPodManager.podOwnerHasNoDelegatedShares(staker))
```
Is more complicated to understand than:
```
eigenPodManager.podOwnerHasDelegatedShares(staker));
```
Similar clarity comment for the stakerStategyListLength check.  It seems like, ideally, this comparison should be
```
return strategyManager.stakerHasDelegatedShares(staker) || eigenPodManager.podOwnerHasDelegatedShares(staker);
```

Reply:
yes, sorry for lack of clarity, this PR to this branch further simplifies this https://github.com/Layr-Labs/eigenlayer-contracts/pull/183